contracts/FragToken.sol

```diff
@@ -70,40 +75,56 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable{
             "Invalid signature"
         );
 
-        // add to current locked amount
-        _locksAmount[msg.sender] = _locksAmount[msg.sender] + amount;
+        _lockInfo.sender = msg.sender;
+        _lockInfo.amount = amount;
+        _lockInfo.unlocked = false;
 
-        if(lock_period == uint256(Period.TwoWeeks))
-            _locktime[msg.sender] = block.timestamp + (2 * _TIMELOCK);
+        if(lock_period == uint256(Period.TwoWeeks)) 
+            _lockInfo.locktime = block.timestamp + (2 * _TIMELOCK);
+        
         
         else if(lock_period == uint256(Period.OneMonth))
-            _locktime[msg.sender] = block.timestamp + (4 * _TIMELOCK);
+            _lockInfo.locktime = block.timestamp + (4 * _TIMELOCK);
+        
 
         else if(lock_period == uint256(Period.ThreeMonths))
-            _locktime[msg.sender] = block.timestamp + (13 * _TIMELOCK);
+            _lockInfo.locktime = block.timestamp + (13 * _TIMELOCK);
+        
 
         else if(lock_period == uint256(Period.SixMonths))
-            _locktime[msg.sender] = block.timestamp + (26 * _TIMELOCK);
+            _lockInfo.locktime = block.timestamp + (26 * _TIMELOCK);
+        
 
         else if(lock_period == uint256(Period.OneYear))
-            _locktime[msg.sender] = block.timestamp + (52 * _TIMELOCK);
+            _lockInfo.locktime = block.timestamp + (52 * _TIMELOCK);
```

Comment:
add an `else` just in case

Reply:
added with a `revert` since it should not happen to fall into that.